lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+      bin - the binary to check for
+
+  DESCRIPTION
+    checkIfBinExists checks to see if the specified binary executable is
+    in the path.
+
+  RETURN VALUE
+    result - true if the executable was found, false if not
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.checkIfBinExists(bin)
+  local handle = io.popen("which "..bin)
+  local result = handle:read()
+  local ret
+  handle:close()
+  if (result) then
+    dt.print_error("true checkIfBinExists: "..bin)
+    ret = true
+  else
+    dt.print_error(bin.." not found")
+    ret = false
+  end
+  return ret
+end
+
+--[[
+  NAME
+    dutils.checkIfFileExists - check if a file or path exist
+
+  SYNOPSIS
+    result = dtutils.checkIfFileExists(filepath)
+      filepath - a file or path to check
+
+  DESCRIPTION
+    checkIfFileExists checks to see if a file or path exists
+
+  RETURN VALUE
+    result - true if the file or path exists, false if it doesn't
+
+  ERRORS
+
+
+  
+]]
+
+-- Thanks Tobias Jakobs for the idea and the correction
+function dtutils.checkIfFileExists(filepath)
+  local file = io.open(filepath,"r")
+  local ret
+  if file ~= nil then 
+    io.close(file) 
+    dt.print_error("true checkIfFileExists: "..filepath)
+    ret = true
+  else 
+    dt.print_error(filepath.." not found")
+    ret = false
+  end
+  return ret
+end
+
+--[[
+  NAME
+    dutils.filename_increment - add a two digit increment to a filename
+
+  SYNOPSIS
+    result = dtutils.filename_increment(filepath)
+      filepath - file to increment
+
+  DESCRIPTION
+    filename_increment solves the problem of filename confllict by adding an 
+    increment to the filename.  If the supplied filename has no increment then 
+    "01" is added to the basename.  If the filename already has an increment, then
+    1 is added to it and the filename returned.
+
+  RETURN VALUE
+    result - the incremented filename
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.filename_increment(filepath)
+
+  -- break up the filepath into parts
+  local path = dtutils.get_path(filepath)
+  local basename = dtutils.get_basename(filepath)
+  local filetype = dtutils.get_filetype(filepath)
+
+  -- check to see if we've incremented before
+  local increment = string.match(basename, "_(%d-)$")
+
+  if increment then
+    -- we do 2 digit increments so make sure we didn't grab part of the filename
+    if string.len(increment) > 2 then
+      -- we got the filename so set the increment to 01
+      increment = "01"
+    else
+      increment = string.format("%02d", tonumber(increment) + 1)
+      basename = string.gsub(basename, "_(%d-)$", "")
+    end
+  else
+    increment = "01"
+  end
+  local incremented_filepath = path .. basename .. "_" .. increment .. "." .. filetype
+
+  dt.print_error("original file was " .. filepath)
+  dt.print_error("incremented file is " .. incremented_filepath)
+
+  return incremented_filepath
+end
+
+--[[
+  NAME
+    dutils.groupIfNotMember - group a new file with an existing one
```

Comment:
If you have an image and duplicate it, those 2 images are grouped.  If you select the duplicate and send it to a processor, then group the result with the duplicate, it breaks the original-duplicate group and creates a duplicate-result group.  This checks to see if the result belongs to a group and if not it checks what group the source image belongs to and groups with it.  So, in the instance above the group would be original-duplicate-result.  Most cases turn out to be group with the original. but this takes care of the edge case where the original is part of a group and not the group leader.